Ginger/Ginger/SolutionWindows/TreeViewItems/NewTreeViewItemBase.cs

```diff
@@ -604,5 +605,44 @@ protected StackPanel NewTVItemFolderHeaderStyle(RepositoryFolderBase repoItemFol
             return stack;
         }
 
+        public override void DeleteAllTreeItems(Type typeOfFolder)
+        {
+            if (Reporter.ToUser(eUserMsgKey.DeleteTreeFolderAreYouSure, mTreeView.Tree.GetSelectedTreeNodeName()) == Amdocs.Ginger.Common.eUserMsgSelection.Yes)
+            {
+                List<ITreeViewItem> childNodes = mTreeView.Tree.GetTreeNodeChildsIncludingSubChilds((ITreeViewItem)this);
+                childNodes.Reverse();
+                foreach (ITreeViewItem node in childNodes)
+                {
+                    if (node == null) continue;
+                    if (node.NodeObject() != null)
+                    {
+                        if (node.NodeObject() is RepositoryFolderBase==false)
```

Comment:
If it is not repository folder not necessary means it is Repository item. Please add condition to check Repository Item base